### ISSUE TITLE
Int64 

### ISSUE THREAD

**Issue by rif** [ericlagergren/decimal#20]
I have another one to report (hope still a quick one). Int64 fails but if we do Int().Int64() than it's OK:

```go
func TestDecInt64(t *testing.T) {
	x := New(10240000000000, 0)
	x.Mul(x, New(976563, 9))
	if x.Int64() != 10000005120 {
		t.Error("error int64: ", x.Int64(), x.Int().Int64())
	}
}
```
```
decimal_test.go:187: error int64:  -844673895 10000005120
```

**Comment by ericlagergren** [comment]
this is expected behavior. Int64 is undefined if x cannot fit into an int64, and the multiplication makes it larger than an int64. I suppose I could (internally) call Int().Int64() if x is larger than an int64...

**Comment by ericlagergren** [comment]
fixed in 1.0
